doc: Bluetooth: Mesh: add settings caches (#13951)
nrfconnect/sdk-nrf: doc/nrf/protocols/bt/bt_mesh/configuring.rst

* Reducing the Node ID advertisement timeout decreases the period where the device consumes power for advertising.

  * :kconfig:option:`CONFIG_BT_MESH_NODE_ID_TIMEOUT` =30.

Persistent storage
------------------

Zephyr's Bluetooth mesh has been designed to use the :ref:`zephyr:settings_api` Settings subsystem to store internal states and options in the :ref:`zephyr:bluetooth_mesh_persistent_storage` persistent memory.
The Settings subsystem can be used with different backends. Mesh is configured with :ref:`zephyr:nvs_api` NVS as the Settings backend.

> However, in general, the Settings subsystem based on NVS is not well designed to store a huge number of states simultaneously

Reviewer: I'd avoid saying that something is not well designed. I'd recommend just to say that when many entries are stored through the settings subsystem (not necessarily RPL), the store time grows significantly. And that users can experience this when using huge RPL.

as well as very often changed data with the default configuration. In the worst scenarios, the storing time might take up to several minutes.
To work quickly with NVS backend, the Settings subsystem requires proper configuration. It is recommended to use internal caches
to improve performance.

* The Settings NVS name cache reduces the number of searching loops of internal parameter identifiers, keeping them in memory.

  * :kconfig:option:`CONFIG_SETTINGS_NVS_NAME_CACHE`.

> * The size of this cache is recommended to be set to the maximum number of devices that the configured device communicates plus the maximum number of often changed model states.

Reviewer: A bit difficult to understand the meaning of this sentence. Perhaps directly mention RPL size here? Can we also suggest some configuration? Give an example for certain use case? Or refer to our samples?

* Reducing the Node ID advertisement timeout decreases the period where the device consumes power for advertising.

  * :kconfig:option:`CONFIG_BT_MESH_NODE_ID_TIMEOUT` =30.

Persistent storage
------------------

Zephyr's Bluetooth mesh has been designed to use the :ref:`zephyr:settings_api` subsystem to store internal states and options in the :ref:`zephyr:bluetooth_mesh_persistent_storage`.
The Settings subsystem can be used with different backends. Mesh is configured with :ref:`zephyr:nvs_api` as the Settings backend.
However, if the Settings subsystem based on NVS stores a huge number of states simultaneously in the worst scenarios,
the storing time might take up to several minutes. The customer can experience that with storing the huge size replay protection list.
To work quickly with NVS backend, the Settings subsystem requires proper configuration. It is recommended to use internal caches
to improve performance.

> * The Settings NVS name cache reduces the number of searching loops of internal parameter identifiers, keeping them in memory.

Reviewer 2: search loops?

  * :kconfig:option:`CONFIG_SETTINGS_NVS_NAME_CACHE`.

> * The size of this cache is recommended to be set to the maximum number of devices those the configured device communicates.

Reviewer 2: ```suggestion
* The size of the Settings NVS name cache is recommended to be set to the maximum number of devices the configured device communicates with.
```

>   For example, if a device communicates with 255 other devices, it is worth setting the cache in 255 minimum.

Reviewer 2: ```suggestion
  For example, if a device communicates with 255 other devices, it is worth setting the cache to minimum 255.
```

  * :kconfig:option:`CONFIG_SETTINGS_NVS_NAME_CACHE_SIZE`.

> * NVS lookup cache reduces the number of searching loops within NVS' application table.

Reviewer 2: search loop?
